lib/src/import_path_base.dart

```diff
@@ -0,0 +1,276 @@
+        return ASCIIArtTreeStyle.dots;
+      case ImportPathStyle.elegant:
+        return ASCIIArtTreeStyle.elegant;
+      default:
+        return null;
+    }
+  }
+}
+
+/// Import Path search tool.
+class ImportPath extends ImportWidget {
+  /// The entry point to start the search.
+  final Uri from;
+
+  /// The import to find. Can be an [Uri] or a [RegExp].
+  /// See [ImportToFind.from].
+  final ImportToFind importToFind;
+
+  /// If `true` searches for all import matches.
+  /// See [ImportPathScanner.findAll].
+  final bool findAll;
+
+  /// If `true` remove from paths the [searchRoot].
+  final bool strip;
+
+  /// If `true`, it will use a fast parser that attempts to
+  /// parse only the import section of Dart files. Default: `false`.
+  /// See [ImportPathScanner.fastParser].
+  final bool fastParser;
+
+  /// If `true`, it will also scan imports that depend on an `if` resolution. Default: `true`.
+  /// See [ImportPathScanner.includeConditionalImports].
+  final bool includeConditionalImports;
+
+  /// The search root to strip from the displayed import paths.
+  /// - If `searchRoot` is not provided at construction it's resolved
+  ///   using [from] parent directory (see [resolveSearchRoot]).
+  late String searchRoot;
+
+  ImportPath(this.from, Object importToFind,
+      {this.findAll = false,
+      bool quiet = false,
+      this.strip = false,
+      this.fastParser = false,
+      this.includeConditionalImports = true,
+      String? searchRoot,
+      MessagePrinter messagePrinter = print})
+      : importToFind = ImportToFind.from(importToFind),
+        super(quiet: quiet, messagePrinter: messagePrinter) {
+    this.searchRoot = searchRoot ?? resolveSearchRoot();
+  }
+
+  /// This list contains common Dart root directories.
+  /// These names are preserved by [resolveSearchRoot] to prevent
+  /// them from being stripped. Default: `'web', 'bin', 'src', 'test', 'example'`
+  List<String> commonRootDirectories = ['web', 'bin', 'src', 'test', 'example'];
+
+  /// Resolves the [searchRoot] using [from] parent.
+  /// See [commonRootDirectories].
+  String resolveSearchRoot() {
+    var rootPath = p.dirname(from.path);
+    var rootDirName = p.split(rootPath).last;
+
+    if (commonRootDirectories.contains(rootDirName)) {
+      var rootPath2 = p.dirname(rootPath);
+      if (rootPath2.isNotEmpty) {
+        rootPath = rootPath2;
+      }
+    }
+
+    var rootUri = from.replace(path: rootPath).toString();
+    return rootUri.endsWith('/') ? rootUri : '$rootUri/';
+  }
+
+  /// Return the search root to [strip] from the displayed import paths.
+  /// If [strip] is `false` returns `null`.
+  /// See [searchRoot].
+  String? get _stripSearchRoot => strip ? searchRoot : null;
+
+  /// Executes the import search and prints the results.
+  /// - [style] defines the output format. Default: [ImportPathStyle.elegant]
+  /// - See [search], [printMessage] and [ASCIIArtTree].
+  Future<ASCIIArtTree?> execute(
```

Comment:
I also feel like this could be an extension method, or just inlined into the `bin/` script.

It is also weird that it both returns the `ASCIIArtTree` and prints out the paths, but not necessarily in ascii format.